diff --git a/gecko/widget/gonk/libdisplay/GonkDisplayJB.cpp b/gecko/widget/gonk/libdisplay/GonkDisplayJB.cpp
index 04caefb..94afcee 100644
--- a/gecko/widget/gonk/libdisplay/GonkDisplayJB.cpp
+++ b/gecko/widget/gonk/libdisplay/GonkDisplayJB.cpp
@@ -216,7 +216,8 @@ GonkDisplayJB::SwapBuffers(EGLDisplay dpy, EGLSurface sur)
     mList->outbufAcquireFenceFd = -1;
 #endif
     eglSwapBuffers(dpy, sur);
-    return Post(mFBSurface->lastHandle, mFBSurface->GetPrevFBAcquireFd());
+    //return Post(mFBSurface->lastHandle, mFBSurface->GetPrevFBAcquireFd());
+    return true;
 }
 
 bool
@@ -307,7 +308,7 @@ GonkDisplayJB::SetFBReleaseFd(int fd)
 int
 GonkDisplayJB::GetPrevFBAcquireFd()
 {
-    return mFBSurface->GetPrevFBAcquireFd();
+    return -1; // mFBSurface->GetPrevFBAcquireFd();
 }
 
 __attribute__ ((visibility ("default")))
