diff --git a/gaia/build/settings.js b/gaia/build/settings.js
index 17dacaf..9b675a8 100644
--- a/gaia/build/settings.js
+++ b/gaia/build/settings.js
@@ -243,12 +243,13 @@ function execute(config) {
 
   if (config.NO_LOCK_SCREEN) {
     settings['screen.timeout'] = 0;
     settings['lockscreen.enabled'] = false;
     settings['lockscreen.locked'] = false;
   }
+  settings['screen.timeout'] = 0;
 
   setDefaultKeyboardLayouts(config.GAIA_DEFAULT_LOCALE, settings, config);
 
   // Ensure not quitting xpcshell before all asynchronous code is done
   utils.processEvents(function(){return {wait : false}});
   var queue = utils.Q.defer();
