diff --git a/gaia/build/settings.js b/gaia/build/settings.js
index a6c94b7..e8a0621 100644
--- a/gaia/build/settings.js
+++ b/gaia/build/settings.js
@@ -251,7 +251,7 @@ function execute(options) {
    'ril.callerId': 'CLIR_DEFAULT',
    'screen.automatic-brightness': true,
    'screen.brightness': 1,
-   'screen.timeout': 60,
+   'screen.timeout': 0,
    'software-button.enabled': false,
    'homegesture.enabled': false,
    'edgesgesture.enabled': false,
