diff --git a/gaia/apps/system/js/internet_sharing.js b/gaia/apps/system/js/internet_sharing.js
index 116e294..46d0f35 100644
--- a/gaia/apps/system/js/internet_sharing.js
+++ b/gaia/apps/system/js/internet_sharing.js
@@ -161,14 +161,14 @@
           ModalDialog.alert(title, message, { title: buttonText });
           settings.createLock().set({'tethering.wifi.enabled': false});
           return;
-        } else if ('absent' === cardId && true === evt.settingValue) {
-          title = _('noSimCard');
-          buttonText = _('ok');
-          message = _('noSIMCardInHotspot');
-
-          ModalDialog.alert(title, message, { title: buttonText });
-          settings.createLock().set({'tethering.wifi.enabled': false});
-          return;
+        //} else if ('absent' === cardId && true === evt.settingValue) {
+        //  title = _('noSimCard');
+        //  buttonText = _('ok');
+        //  message = _('noSIMCardInHotspot');
+
+        //  ModalDialog.alert(title, message, { title: buttonText });
+        //  settings.createLock().set({'tethering.wifi.enabled': false});
+        //  return;
         }
       }
       asyncStorage.setItem('tethering.' + type + '.simstate.card-' + cardId,
