diff --git a/gaia/Makefile b/gaia/Makefile
index c026714..bff988a 100644
--- a/gaia/Makefile
+++ b/gaia/Makefile
@@ -296,12 +296,16 @@ export SYS
 export GAIA_DIR
 export SEP
 export SEP_FOR_SED
 
 ifndef GAIA_APP_CONFIG
 GAIA_APP_CONFIG=$(GAIA_DIR)$(SEP)build$(SEP)config$(SEP)$(GAIA_DEVICE_TYPE)$(SEP)apps-$(GAIA_APP_TARGET).list
+ifeq ($(wildcard $(GAIA_APP_CONFIG)),)
+# fallback to use engineering app list, since some device type, such as tv, does not have production list
+GAIA_APP_CONFIG=$(GAIA_DIR)$(SEP)build$(SEP)config$(SEP)$(GAIA_DEVICE_TYPE)$(SEP)apps-engineering.list
+endif
 endif
 
 ifndef GAIA_DISTRIBUTION_DIR
   GAIA_DISTRIBUTION_DIR := $(GAIA_DIR)$(SEP)distribution
 else
 	ifneq (,$(findstring MINGW32_,$(SYS)))
