diff --git a/gaia/build/settings.js b/gaia/build/settings.js
index 58e5b4e..f1cd0f6 100644
--- a/gaia/build/settings.js
+++ b/gaia/build/settings.js
@@ -221,12 +221,13 @@ function execute(config) {
 
     // Activate developer menu under the system menu when long pressing
     // the power button by default for devs.
     settings['developer.menu.enabled'] = true;
   }
 
+  settings['device.storage.writable.name'] = 'internal_sd';
   // Set the ftu manifest URL
   if (config.NOFTU === '0') {
     settings['ftu.manifestURL'] = utils.gaiaManifestURL('ftu',
       config.GAIA_SCHEME, config.GAIA_DOMAIN, config.GAIA_PORT);
   }
 
