diff --git a/gecko/gfx/layers/opengl/GrallocTextureHost.cpp b/gecko/gfx/layers/opengl/GrallocTextureHost.cpp
index 73ac05c..674fa7b 100644
--- a/gecko/gfx/layers/opengl/GrallocTextureHost.cpp
+++ b/gecko/gfx/layers/opengl/GrallocTextureHost.cpp
@@ -33,12 +33,13 @@ SurfaceFormatForAndroidPixelFormat(android::PixelFormat aFormat,
     return swapRB ? gfx::SurfaceFormat::B8G8R8X8 : gfx::SurfaceFormat::R8G8B8X8;
   case android::PIXEL_FORMAT_RGB_565:
     return gfx::SurfaceFormat::R5G6B5;
   case HAL_PIXEL_FORMAT_YCbCr_422_SP:
   case HAL_PIXEL_FORMAT_YCrCb_420_SP:
   case HAL_PIXEL_FORMAT_YCbCr_422_I:
+  case HAL_PIXEL_FORMAT_YCrCb_NV12:
   case GrallocImage::HAL_PIXEL_FORMAT_YCbCr_420_SP_TILED:
   case GrallocImage::HAL_PIXEL_FORMAT_YCbCr_420_SP_VENUS:
   case HAL_PIXEL_FORMAT_YV12:
     return gfx::SurfaceFormat::R8G8B8A8; // yup, use SurfaceFormat::R8G8B8A8 even though it's a YUV texture. This is an external texture.
   default:
     if (aFormat >= 0x100 && aFormat <= 0x1FF) {
