diff --git a/gecko/content/media/DecoderTraits.cpp b/gecko/content/media/DecoderTraits.cpp
index d4c191d..4f77013 100644
--- a/gecko/content/media/DecoderTraits.cpp
+++ b/gecko/content/media/DecoderTraits.cpp
@@ -223,6 +223,17 @@ static const char* const gOmxTypes[] = {
   "video/3gpp",
   "video/3gpp2",
   "video/quicktime",
+  "video/x-flv",
+  "video/x-f4v",
+  "video/x-msvideo",
+  "video/x-matroska",
+  "video/mp2t",
+  "video/mp2p",
+  "application/octet-stream",
+  "application/x-mpegURL",
+  "application/x-mpegurl",
+  "application/vnd.apple.mpegURL",
+  "application/vnd.apple.mpegurl",
 #ifdef MOZ_OMX_WEBM_DECODER
   "video/webm",
   "audio/webm",
