diff --git a/gecko/widget/gonk/libdisplay/FramebufferSurface.cpp b/gecko/widget/gonk/libdisplay/FramebufferSurface.cpp
index 8d9b239..c826015 100644
--- a/gecko/widget/gonk/libdisplay/FramebufferSurface.cpp
+++ b/gecko/widget/gonk/libdisplay/FramebufferSurface.cpp
@@ -33,12 +33,13 @@
 #include <gui/SurfaceTextureClient.h>
 #endif
 #include <ui/GraphicBuffer.h>
 
 #include "FramebufferSurface.h"
 #include "GraphicBufferAlloc.h"
+#include "GonkDisplayJB.h"
 
 #ifndef NUM_FRAMEBUFFER_SURFACE_BUFFERS
 #define NUM_FRAMEBUFFER_SURFACE_BUFFERS (2)
 #endif
 
 // ----------------------------------------------------------------------------
@@ -138,12 +139,15 @@ void FramebufferSurface::onFrameAvailable() {
     if (acquireFence.get() && acquireFence->isValid())
         mPrevFBAcquireFence = new Fence(acquireFence->dup());
     else
         mPrevFBAcquireFence = Fence::NO_FENCE;
 
     lastHandle = buf->handle;
+    extern mozilla::GonkDisplay* GetGonkDisplay();
+
+    (static_cast<mozilla::GonkDisplayJB*>(mozilla::GetGonkDisplay()))->Post(lastHandle, -1);
 }
 
 void FramebufferSurface::freeBufferLocked(int slotIndex) {
     ConsumerBase::freeBufferLocked(slotIndex);
     if (slotIndex == mCurrentBufferSlot) {
         mCurrentBufferSlot = BufferQueue::INVALID_BUFFER_SLOT;
