diff --git a/gecko//configure.in b/gecko/configure.in
index 2ae88db..f53e15b 100644
--- a/gecko/configure.in
+++ b/gecko/configure.in
@@ -270,6 +270,7 @@ if test -n "$gonkdir" ; then
         MOZ_B2G_BT_BLUEDROID=1
         MOZ_NFC=1
         MOZ_RTSP=1
+        NECKO_PROTOCOL_rtsp=1
         MOZ_OMX_DECODER=1
         MOZ_OMX_ENCODER=1
         AC_DEFINE(MOZ_OMX_ENCODER)
@@ -3823,8 +3824,12 @@ NSS_DISABLE_DBM=
 NECKO_COOKIES=1
 NECKO_PROTOCOLS_DEFAULT="about app data file ftp http res viewsource websocket wyciwyg device"
 if test -n "$MOZ_RTSP"; then
+  AC_DEFINE(MOZ_RTSP)
+  AC_DEFINE(NECKO_PROTOCOL_rtsp)
   NECKO_PROTOCOLS_DEFAULT="$NECKO_PROTOCOLS_DEFAULT rtsp"
 fi
+AC_SUBST(NECKO_PROTOCOL_rtsp)
+AC_SUBST(MOZ_RTSP)
 USE_ARM_KUSER=
 BUILD_CTYPES=1
 MOZ_USE_NATIVE_POPUP_WINDOWS=
